extensions/schema-registry/confluent/common/deployment/src/main/java/io/quarkus/confluent/registry/common/ConfluentRegistryClientProcessor.java

```diff
@@ -19,9 +20,12 @@ public void confluentRegistryClient(
         if (curateOutcomeBuildItem.getApplicationModel().getDependencies().stream().anyMatch(
                 x -> x.getGroupId().equals("io.confluent")
                         && x.getArtifactId().equals("kafka-schema-serializer"))) {
-            reflectiveClass
-                    .produce(new ReflectiveClassBuildItem(true, false, false,
-                            "io.confluent.kafka.serializers.context.NullContextNameStrategy"));
+
+            String nullContextNameStrategy = "io.confluent.kafka.serializers.context.NullContextNameStrategy";
+            if (QuarkusClassLoader.isClassPresentAtRuntime(nullContextNameStrategy)) {
+                // Class not present before v7.0.0, the check may be removed once apicurio 1.x support is dropped
```

Comment:
This has nothing to do with Apicurio, does it? It's just that we need to distinguish between older and newer Confluent libraries. I think I'd just

```suggestion
                // Class not present before v7.0.0
```

I wonder how come this wasn't a problem before, but other than that, it seems fine.

Reply:
> This has nothing to do with Apicurio, does it? It's just that we need to distinguish between older and newer Confluent libraries.

You are right this is only related to Apicurio in the sense that versions < v7 are only tested in `integration-tests/kafka-avro` where apicurio 1.x is being tested.  

> I wonder how come this wasn't a problem before, but other than that, it seems fine.

Missing classes are currently silently ignored so it was not an issue.